packages/core/src/browser/label-provider.ts

```diff
@@ -154,6 +196,122 @@ export class DefaultUriLabelProviderContribution implements LabelProviderContrib
+    }
+
+    // copied and modified from https://github.com/microsoft/vscode/blob/1.44.2/src/vs/workbench/services/label/common/labelService.ts#L240-L274
+    /*---------------------------------------------------------------------------------------------
+    *  Copyright (c) Microsoft Corporation. All rights reserved.
+    *  Licensed under the MIT License. See License.txt in the project root for license information.
+    *--------------------------------------------------------------------------------------------*/
+    private readonly labelMatchingRegexp = /\${(scheme|authority|path|query)}/g;
+    private formatUri(resource: URI, formatting: ResourceLabelFormatting): string {
+        // const labelMatchingRegexp = /\${(scheme|authority|path|query)}/g;
+        let label = formatting.label.replace(this.labelMatchingRegexp, (match, token) => {
+            switch (token) {
+                case 'scheme': return resource.scheme;
+                case 'authority': return resource.authority;
+                case 'path': return resource.path.toString();
+                case 'query': return resource.query;
+                default: return '';
+            }
+        });
+
+        // convert \c:\something => C:\something
+        if (formatting.normalizeDriveLetter && this.hasDriveLetter(label)) {
+            label = label.charAt(1).toUpperCase() + label.substr(2);
+        }
+
+        if (formatting.tildify) {
+            const homePath = process.env.HOMEPATH;
+            label = this.tildify(label, homePath ? homePath : '');
+        }
+        if (formatting.authorityPrefix && resource.authority) {
+            label = formatting.authorityPrefix + label;
+        }
+
+        return label.replace(/\//g, formatting.separator);
+    }
+
+    // copied and modified from https://github.com/microsoft/vscode/blob/1.44.2/src/vs/workbench/services/label/common/labelService.ts#L72-L74
+    /*---------------------------------------------------------------------------------------------
+    *  Copyright (c) Microsoft Corporation. All rights reserved.
+    *  Licensed under the MIT License. See License.txt in the project root for license information.
+    *--------------------------------------------------------------------------------------------*/
+    private hasDriveLetter(path: string): boolean {
+        return !!(path && path[2] === ':');
+    }
+
+    // copied and modified from https://github.com/microsoft/vscode/blob/1.44.2/src/vs/base/common/labels.ts#L107-L125
+    /*---------------------------------------------------------------------------------------------
+    *  Copyright (c) Microsoft Corporation. All rights reserved.
+    *  Licensed under the MIT License. See License.txt in the project root for license information.
+    *--------------------------------------------------------------------------------------------*/
+    private tildify(path: string, userHome: string): string {
+        if (isWindows || !path || !userHome) {
```

Comment:
`isWindows` check is meaningless in the browser, operating system of backend can be Linux

We should use `EnvVarServer` to expose backend operating system and use it here.

Reply:
Moved the `tildifyPath()` from `filesystem` to `core` and used it in the `DefaultUriLabelProviderContribution`